### PR TITLE
Add pixel-caffiene/frontend.js to auto-delay list

### LINKED ISSUE #3088: Delay JavaScript - Pixel Caffeine is breaking
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version ✅
- Used the search feature to ensure that the bug hasn’t been reported before ✅

**Describe the bug**

When the Pixel Caffeine plugin is used along with the **Delay JavaScript scripts** feature, there is a JavaScript console error:
```
 aepc_extend_args is not defined
```

This is because they are using `fbq( 'init' )` and we automatically delay `fbq(`.

**To Reproduce**
1. Install/Set up the [Pixel Caffeine](https://wordpress.org/plugins/pixel-caffeine/) plugin.
2. Enable **Delay JavaScript scripts**
3. Check the JavaScript console for the error.

**Expected behavior**

There shouldn't be any errors in the JavaScript console.

**Screenshots**
screenshot

**Additional context**

Delaying the following script fixes the error:
```
/wp-content/plugins/pixel-caffeine/build/frontend.js
```

**Related Ticket:** https://secure.helpscout.net/conversation/1279175035/193966/

**Backlog Grooming (for WP Media dev team use only)**
- [x] Reproduce the problem
- [x] Identify the root cause
- [x] Scope a solution
- [x] Estimate the effort

### PR DESCRIPTION
**Pull request by iCaspar**
Closes #3088.

- [X] Add caffeine frontend.js to default delayed scripts.
- [X] Update test fixtures.

### CODE REVIEW

**Review by remyperona — CHANGES_REQUESTED** [review]
In addition, you need to add a new method to hook on the upgrade routine, to add this new script to the saved user list in the settings

**Comment by iCaspar** [comment]
Modified the current set_options_on_update() to check for the new script and add if it's not already there. Ready for another look @Tabrisrp .

**Review by remyperona — APPROVED** [review]

**Comment by iCaspar** [comment]
@wp-media/qa This one is ready for your magic!

**Review by remyperona — APPROVED** [review]

**Comment by Tabrisrp** [comment]
@wp-media/qa Ready for QA

- The pixel-caffeine script should be delayed too
- The script path should be added automatically to the default list
- The script path should be added to the saved value in the option if the `fbq` pattern is also present, but not if it's not

**Comment by iCaspar** [comment]
@Tabrisrp Just a last quick check on the integration tests I added there. QA can continue during review.

File: tests/Integration/inc/Engine/Optimization/DelayJS/Admin/Subscriber/optionUpdate372.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+		parent::tearDown();
+
+		$this->restoreWpFilter( 'wp_rocket_upgrade' );
+	}
+
+	/**
+	 * @dataProvider configTestData
+	 */
+	public function testShouldDoExpected( $config, $expected ) {
+		global $wp_filter;
```
Comment (Tabrisrp) [inline tests/Integration/inc/Engine/Optimization/DelayJS/Admin/Subscriber/optionUpdate372.php:29]:
```suggestion
```

File: tests/Integration/inc/Engine/Optimization/DelayJS/Admin/Subscriber/optionUpdate372.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+namespace WP_Rocket\Tests\Integration\inc\Engine\Optimization\DelayJS\Admin\Subscriber;
+
+use WP_Rocket\Tests\Integration\TestCase;
+
+/**
+ * @covers \WP_Rocket\Engine\Optimization\DelayJS\Admin\Subscriber::option_update_3_7_2
+ *
+ * @group  DelayJS
+ */
+class Test_optionUpdate372 extends TestCase{
```
Comment (Tabrisrp) [inline tests/Integration/inc/Engine/Optimization/DelayJS/Admin/Subscriber/optionUpdate372.php:12]:
```suggestion
class Test_optionUpdate372 extends TestCase {
```

File: tests/Unit/inc/Engine/Optimization/DelayJS/Admin/Settings/optionUpdate372.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+use WP_Rocket\Admin\Options_Data;
+use WP_Rocket\Engine\Optimization\DelayJS\Admin\Settings;
+use WP_Rocket\Tests\Unit\TestCase;
+
+/**
+ * @covers \WP_Rocket\Engine\Optimization\DelayJS\Admin\Settings::option_update_3_7_2
+ *
+ * @group  DelayJS
+ */
+class Test_OptionUpdate372 extends TestCase{
```
Comment (Tabrisrp) [inline tests/Unit/inc/Engine/Optimization/DelayJS/Admin/Settings/optionUpdate372.php:15]:
```suggestion
class Test_OptionUpdate372 extends TestCase {
```

**Review by remyperona — APPROVED** [review]
just minor formatting

**Review by piotrbak — APPROVED** [review]
Worked as expected, tested scenarios:
- Should add the new string to the default list during the clean installation
- Should add the new string to the default list after the update when the list was default one
- Should not add the new string to the default list after the update when the string was already added there
- Should not add pixel caffeine script if fbq( script is not present on the list
- Should validate Delay JS is compatible with pixel caffeine

**Comment by antoscarface** [comment]
Hi, I'm the Pixel Caffeine developer :) Just a heads up: first of all thank you for adding this into WP Rocket. Secondly, I've already fixed my self this into the latest version of Pixel Caffeine (before to notice this PR, I'm seeing it only now!) by splitting the main `<script>`, which include global variables definition and the Facebook Pixel snippet initialization, loading them separately, so WP Rocket will delay the snippet init, but not the global variable's definition, in order to prevent the `aepc_something not found` error.

Anyway, that's great you added directly the frontend.js file, it makes sense as well.
Thank you :)

**Pull request merged**